internal/js/modules/k6/browser/common/network_manager.go

```diff
@@ -895,32 +902,69 @@ func (m *NetworkManager) ContinueRequest(requestID fetch.RequestID) {
 		if strings.Contains(err.Error(), "Invalid InterceptionId") {
 			m.logger.Debugf("NetworkManager:ContinueRequest", "invalid interception ID (%s) continuing request: %s",
 				requestID, err)
-			return
+			return nil
 		}
 
-		m.logger.Errorf("NetworkManager:ContinueRequest", "fail to continue request (id: %s): %s",
-			requestID, err)
+		return fmt.Errorf("fail to continue request (id: %s): %w", requestID, err)
 	}
+
+	return nil
 }
 
-func (m *NetworkManager) FulfillRequest(requestID fetch.RequestID, params fetch.FulfillRequestParams) {
-	action := fetch.FulfillRequest(requestID, params.ResponseCode).
-		WithResponseHeaders(params.ResponseHeaders).
-		WithResponsePhrase(params.ResponsePhrase).
-		WithBody(params.Body)
+func (m *NetworkManager) FulfillRequest(request *Request, opts FulfillOptions) error {
+	responseCode := int64(http.StatusOK)
+	if opts.Status != 0 {
+		responseCode = opts.Status
+	}
+
+	action := fetch.FulfillRequest(request.interceptionID, responseCode)
+
+	if opts.ContentType != "" {
```

Comment:
I'm wondering here... Is it better to keep it like this or actually remove this field from the `FulfillOptions` struct and directly add it to the `Headers` option field in `parseFulfillOptions` method, WDYT?

Reply:
Yes I agree, this would only be "under the hood". The API spec would stay the same but we wouldn't bother with this field in the other layers of the codebase.